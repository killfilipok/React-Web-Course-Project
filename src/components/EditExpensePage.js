import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
    startEditExpense = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/')
    }

    startRemoveExpense = () => {
        this.props.startRemoveExpense(this.props.expense.id)
        // .then(() => {
        //     this.props.history.push('/')
        // })
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>

                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.startEditExpense}
                    />

                    <button
                        className="button--grey"
                        onClick={this.startRemoveExpense}
                    >
                        Remove Expense
                    </button>
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expenseToEdit) => dispatch(startEditExpense(id, expenseToEdit)),
    startRemoveExpense: (id) => { return dispatch(startRemoveExpense({ id })) }
})

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)