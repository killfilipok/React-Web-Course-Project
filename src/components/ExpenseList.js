import React from 'react'
import { connect } from 'react-redux'
import ExpenseItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div className="content-container">
        <div>
            <div>Expense</div>
            <div>Amount</div>
        </div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                    props.expenses.map((expense, index) =>
                        <ExpenseItem
                            key={expense.id}
                            {...expense}
                        />
                    )
                )
        }
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList)

