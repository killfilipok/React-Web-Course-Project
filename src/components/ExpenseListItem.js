import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

export default ({ description, amount, createdAt, id }) => (
    <div>
        <Link to={`/edit/${id}`}>Expense List Item: </Link>
        <p>
             {description} 
            - 
            {numeral(amount /100).format('$0,0.00')} 
            created at: {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
)

