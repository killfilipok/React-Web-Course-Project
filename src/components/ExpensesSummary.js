import React from 'react'
import { connect } from 'react-redux'
import SelectExpensesTotal from './SelectExpensesTotal'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
    const total = numeral(expensesTotal / 100).format('$0,0.00')
    const textToRender = `viewing ${expensesCount} expense${expensesCount>1?'s' : ''} totaling: ${total}`

    return (
        <div>
            <h3>{textToRender}</h3>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: SelectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)