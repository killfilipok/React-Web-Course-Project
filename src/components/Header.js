import React from 'react'
import { startLogOut } from '../actions/auth'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'

export const Header = ({ startLogOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="is-active" >Home shity home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create page</NavLink>
        <button onClick={startLogOut}>Logout</button>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
})
export default connect(undefined, mapDispatchToProps)(Header)