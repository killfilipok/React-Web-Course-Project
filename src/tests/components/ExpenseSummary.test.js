import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import {shallow} from 'enzyme'
import React from 'react'

test('should return total count and amount for no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={0} expensesTotal={0}/>)

    expect(wrapper).toMatchSnapshot()
})


test('should return total count and amount', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={15000}/>)

    expect(wrapper).toMatchSnapshot()
})


test('should return total count and amount fro one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1511}/>)

    expect(wrapper).toMatchSnapshot()
})
