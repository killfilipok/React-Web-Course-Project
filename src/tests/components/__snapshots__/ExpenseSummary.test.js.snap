// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return total count and amount 1`] = `
<div>
  <h3>
    viewing 3 expenses totaling: $150.00
  </h3>
</div>
`;

exports[`should return total count and amount for no expenses 1`] = `
<div>
  <h3>
    viewing 0 expense totaling: $0.00
  </h3>
</div>
`;

exports[`should return total count and amount fro one expense 1`] = `
<div>
  <h3>
    viewing 1 expense totaling: $15.11
  </h3>
</div>
`;
